File: 24-Jan-2021/Daily Challenges/Question-1 solution.py
def reverseArray(given, flag):
    if(flag):
        # USING INBUILT METHOD
        given.reverse()
    else:
        # IMPLEMENTING ON MY OWN
        for i in range((len(given)-1)/2):
            temp = given[i]
            given[i] = given[len(given)-1]
            given[len(given)-1] = temp


if __name__ == "__main__":
    given = input().split()
    reverseArray(given, True)
    print(" ".join(given))

File: 24-Jan-2021/Daily Challenges/Question-1 solution.cpp
#include <iostream>
#include <algorithm>

using namespace std;


void reverseArray(int P[], int Q) {
    // USING INBUILT METHOD
    // reverse(P, P+Q);
    
    // IMPLEMENTING IT
    for(int i=0; i<=(Q-1)/2; i++) {
        int temp = P[i];
        P[i] = P[Q-1-i];
        P[Q-1-i] = temp;
    }
}


void printArray(int P[], int Q) {
    for(int i=0; i<Q; i++) {
        cout<<P[i]<<"\n";
    }
}


int main() {
    int Q;
    cin>>Q;
    
    int P[Q];
    
    for(int i=0; i<Q; i++) {
        cin>>P[i];
    }
    
    reverseArray(P, Q);
    printArray(P, Q);
}